src/us/kbase/workspace/database/Workspace.java

```diff
@@ -918,32 +920,33 @@ private ValidatedTypedObject validate(
 			final IdReferenceHandlerSet<IDAssociation> idhandler,
 			final int objcount)
 			throws TypeStorageException, TypedObjectSchemaException,
-			TypedObjectValidationException {
+				TypedObjectValidationException {
 		final ValidatedTypedObject rep;
 		try {
 			rep = validator.validate(wo.getData(), wo.getType(), idhandler);
 		} catch (NoSuchTypeException nste) {
 			throw new TypedObjectValidationException(String.format(
-					"Object %s failed type checking:\n",
-					getObjectErrorId(wo, objcount))
+					"Object %s failed type checking:\n", getObjectErrorId(wo, objcount))
 					+ nste.getLocalizedMessage(), nste);
 		} catch (NoSuchModuleException nsme) {
 			throw new TypedObjectValidationException(String.format(
-					"Object %s failed type checking:\n",
-					getObjectErrorId(wo, objcount))
+					"Object %s failed type checking:\n", getObjectErrorId(wo, objcount))
 					+ nsme.getLocalizedMessage(), nsme);
 		} catch (TooManyIdsException e) {
-			throw wrapTooManyIDsException(objcount, idhandler, e);
+			throw wrapTooManyIDsException(wo, objcount, idhandler.getMaximumIdCount(), e);
 		} catch (JsonParseException jpe) {
 			throw new TypedObjectValidationException(String.format(
-					"Object %s failed type checking ",
-					getObjectErrorId(wo, objcount)) + 
-					"- a fatal JSON processing error occurred: "
-					+ jpe.getMessage(), jpe);
+					"Object %s failed type checking ", getObjectErrorId(wo, objcount)) + 
+					"- a fatal JSON processing error occurred: " + jpe.getMessage(), jpe);
 		} catch (IOException ioe) {
 			throw new TypedObjectValidationException(String.format(
-					"A fatal IO error occured while type checking object %s: ",
-					getObjectErrorId(wo, objcount)) + ioe.getMessage(), ioe);
+					"Object %s failed type checking ", getObjectErrorId(wo, objcount)) + 
+					"- a fatal IO error occurred: " + ioe.getMessage(), ioe);
+		} catch (TypeFetchException tfe) {
+			throw new TypedObjectValidationException(String.format(
+					"Object %s failed type checking ", getObjectErrorId(wo, objcount)) + 
```

Comment:
Hmm, now I'm thinking I should change `getObjectErrorId` to `getExceptionPrefix` and include the first string since it's always the same for this block

Reply:
could always put that in the next PR

Reply:
done, although it's more like 10 PRs down the line